plugins/HeightProfile.jsx

```diff
@@ -300,19 +300,23 @@ class HeightProfile extends React.Component {
                 nodes.unshift({x: x[0], y: y[0]});
                 nodes.push({x: x[x.length - 1], y: y[y.length - 1]});
 
-                return {
-                    dataset: entry.dataset,
-                    x: x,
-                    y: elevations,
-                    maxY: Math.max(...elevations),
-                    totLength: totLength,
-                    nodes: nodes
-                };
-            });
+                if (!elevations.every(val => val === 0)) {
```

Comment:
Since you are computing `Math.min(...elevations.filter(elev => elev !== 0))` below, you might as well do
```
nonZeroElevations = elevations.filter(elev => elev !== 0)
if (nonZeroElevations.length > 0) {
...
minY: Math.min(...nonZeroElevations),
...
}
```

Saves off one iteration over the data.